app/controllers/responses_controller.rb

```diff
@@ -68,6 +71,21 @@ def response_params
       params.require(:response).permit(answers_attributes: [{raw: []}, :raw, :answerable_type, :answerable_id])
     end
 
+    def validate_responses(params)
```

Comment:
Can you explain this method to me?

Reply:
Sure, this method will either return the params, if the required questions have all been answered or nil if there are any issues - so missing answered questions. In reality, due to the frontend validations there is no way someone could possibly submit the wrong questions through the website itself. So this is more of a protection against for example bugs elsewhere.

Reply:
Hmm ok, I think we could make this a little simpler, for example, we could raise the error and add it to the Response object, which would be a much more rails way of rejecting invalid params. I know we might be pressed for time so I can merge this if you like, and we can come back to it later on perhaps?

Reply:
Please don't merge this yet! I will try and rework this to make it better.